pymc/tests/test_sampling_jax.py

```diff
@@ -270,3 +275,47 @@ def test_seeding(chains, random_seed, sampler):
     if chains > 1:
         assert np.all(result1.posterior["x"].sel(chain=0) != result1.posterior["x"].sel(chain=1))
         assert np.all(result2.posterior["x"].sel(chain=0) != result2.posterior["x"].sel(chain=1))
+
+
+@pytest.mark.parametrize(
+    "nuts_kwargs",
+    [
+        {"adapt_step_size": False},
+        {"adapt_mass_matrix": True},
+        {"dense_mass": True},
+        {"adapt_step_size": False, "adapt_mass_matrix": True, "dense_mass": True},
+        {"fake-key": "fake-value"},
+    ],
+)
+def test_update_numpyro_nuts_kwargs(nuts_kwargs: Dict[str, Any]):
+    original_kwargs = nuts_kwargs.copy()
+    new_kwargs = _update_numpyro_nuts_kwargs(nuts_kwargs)
+
+    # Maintains original key-value pairs.
+    for k, v in original_kwargs.items():
+        assert new_kwargs[k] == v
+
+    for k, v in _numpyro_nuts_defaults().items():
+        if k not in original_kwargs:
+            assert new_kwargs[k] == v
+
+
+@pytest.mark.parametrize(
+    "nuts_kwargs",
+    [
+        {"adapt_step_size": False},
+        {"adapt_mass_matrix": True},
+        {"dense_mass": True},
+        {"adapt_step_size": False, "adapt_mass_matrix": True, "dense_mass": True},
+        {"adapt_step_size": False, "step_size": 0.13},
+    ],
+)
```

Comment:
Let's try to monkey patch? If it's not possible I think we can survive without explicit tests. This one seems too inneficient.

Reply:
I'll give it shot. I agree that the current method I used is inefficient and isn't really testing the desired behavior.